src/pages/index.tsx

```diff
@@ -1,10 +1,49 @@
 import React from 'react';
 import { graphql } from 'gatsby';
 import Index from '../templates/index';
+import path from 'path';
+
+const findNodes = (keyToFind, props, valueToFind = null) => {
+  if (typeof valueToFind === 'string') {
+    return props.data.allMarkdownRemark.edges.filter(
+      (edge) => edge.node.frontmatter[keyToFind] === valueToFind,
+    ).map(edge => {
+      return {
+        ...edge.node,
+        ...edge.node.frontmatter,
+      };
+    });
+  } else {
+    return props.data.allMarkdownRemark.edges.filter(
+      (edge) => path.basename(edge.node.fileAbsolutePath) === keyToFind,
+    );
+  }
+};
+
+const findNode = (keyToFind, props) => {
+  const elements = findNodes(keyToFind, props);
+  return elements[0].node;
+};
```

Comment:
These functions would likely be moved into a shared library, but popped them here for now just so I can tinker with them.